Workspace settings page updates

siovan-ssc opened the pull request:
Added functionality to WorkspaceSettingsPage:
- Workspace leads can update workspace budgets and delete workspaces
- Admins can see the settings but not make any changes (fields are disabled)
- Other user types cannot see these settings

Updated DatahubAuthView so workspace leads are also required to have a project acronym specified.

Added basic .feature file for testing of WorkspaceSettingsPage.

siovan-ssc committed:
Workspace settings updates

Added functionality so workspace leads can update workspace budgets and delete workspaces and admins can see the settings but not make any changes.

yjmrobert commented on Portal/src/Datahub.Portal/Pages/Workspace/Settings/WorkspaceSettingsPage.razor:
```diff
@@ -51,16 +86,46 @@
 
         _budget = project.Project_Budget ?? 0;
+
+        var currentUser = await _userInformationService.GetCurrentPortalUserAsync();
+        var _projectUsers = await _projectUserManagementService.GetProjectUsersAsync(WorkspaceAcronym);
+
+        foreach (var user in _projectUsers)
+        {
+            if (user.Role.Id.Equals(2) && user.PortalUser.Id.Equals(currentUser.Id))
+            {
+                _isDisabled = false;
+            }
```
That's the right idea, but I'll show you the `NotAuthorized` section in the Auth view that'll simplify things

yjmrobert reviewed (changes requested):
Nice work, I like the update in progress UX! Do you happen to have a screenshot of it?


Let's update the disabled to work off the user roles via the NotAuthorized render component.

All that's left is to send the new Workspace Definition onto the queue. You might be able to reuse a method in one of the project management services. 

I'm more than happy to jump on a call to explain some of the topology as there are a lot of moving parts

siovan-ssc committed:
PR updates

Disabled now works off NotAuthorized; added _requestManagementService await.

siovan-ssc commented:
Updated WorkspaceSettingsPage:
- Disabled works off Authorized / NotAuthorized components
- Added _requestManagementService await for sending to the queue

Also, here is what the UX looks like:
When budget has been changed:
![Screenshot 2024-03-11 153819](https://github.com/ssc-sp/datahub-portal/assets/67435074/e181d815-6051-4f8c-ad70-6595ca5dcaa1)

When saving the new budget:
![Screenshot 2024-03-11 153804](https://github.com/ssc-sp/datahub-portal/assets/67435074/30f18f2f-12a7-45f5-831d-3343ffd272c7)

yjmrobert committed:
📜 added localization

yjmrobert committed:
🔧 Update Workspace Settings authorization and layout

Changed the authorization level for modifying the Workspace settings from WorkspaceLead to DatahubAdmin. Also revised the layout of the settings page for better visibility of save changes alert. Snackbars are now displayed when changes are saved or if there are no changes to be saved.

yjmrobert committed:
Merge branch 'develop' into workspace_settings_branch

yjmrobert commented:
Looks good! Great work on the UI/UX. I did a minor update to only allow datahub admins instead of workspace leads (just until we figure out how costing is going to work in production).

Cheers!
